src/main/java/ch/wisv/areafiftylan/controller/ConsumptionController.java

```diff
@@ -0,0 +1,81 @@
+package ch.wisv.areafiftylan.controller;
+
+import ch.wisv.areafiftylan.dto.ConsumptionDTO;
+import ch.wisv.areafiftylan.exception.AlreadyConsumedException;
+import ch.wisv.areafiftylan.exception.ConsumptionNotFoundException;
+import ch.wisv.areafiftylan.model.util.Consumption;
+import ch.wisv.areafiftylan.service.ConsumptionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import static ch.wisv.areafiftylan.util.ResponseEntityBuilder.createResponseEntity;
+
+/**
+ * Created by beer on 16-5-16.
+ */
+@RestController
+@PreAuthorize("hasRole('ADMIN')")
+@RequestMapping(value = "/consumptions")
+public class ConsumptionController {
+    @Autowired
+    private ConsumptionService consumptionService;
+
+    @RequestMapping(method = RequestMethod.GET)
+    public Map<String, Boolean> isConsumed(@RequestBody ConsumptionDTO consumptionDTO){
+        boolean consumed = consumptionService.isConsumed(consumptionDTO.getTicketId(), consumptionDTO.getConsumptionId());
+        return Collections.singletonMap("consumed", consumed);
+    }
+
+    @RequestMapping(value = "/{ticketId}", method = RequestMethod.GET)
+    public Collection<Consumption> consumptionsMade(@PathVariable Long ticketId){
+        return consumptionService.getByTicketIdIfValid(ticketId).getConsumptionsMade();
+    }
+
+    @RequestMapping(value = "/available", method = RequestMethod.GET)
+    public Collection<Consumption> getAllPossibleConsumptions(){
+        return consumptionService.getPossibleConsumptions();
+    }
+
+    @RequestMapping(value = "/available", method = RequestMethod.POST)
+    public ResponseEntity<?> addAvailableConsumption(@RequestBody String consumptionName){
+        consumptionService.addPossibleConsumption(consumptionName);
+
+        return createResponseEntity(HttpStatus.OK, "Successfully added " + consumptionName + " as a supported consumption.");
+    }
+
+    @RequestMapping(value = "/available", method = RequestMethod.DELETE)
+    public ResponseEntity<?> removeAvailableConsumption(@RequestBody Long consumptionId){
+        Consumption c = consumptionService.removePossibleConsumption(consumptionId);
+
+        return createResponseEntity(HttpStatus.OK, "Successfully removed " + c.getName() + " as a supported consumption.");
+    }
+
+    @RequestMapping(value = "/consume", method = RequestMethod.POST)
```

Comment:
Could you elaborate why it's better to pass parameters as a PathVariable instead of a DTO? I don't really see the difference, and prefer having all parameters in a single object instead of fragmented over PathVariable/RequestBody.